### PR TITLE
Switch to field tuples for default fields in particle_trajectories

### PR DESCRIPTION
**Pull request by mtryan83**
## PR Summary

This PR changes two things:

1. Default fields `'particle_position_x/y/z'` now added to field_data through `fds` proper name. Since `fds[field]` is created through `dd_first._determine_fields((self.ptype, field))[0]`, this should ensure the field exists and is a tuple with the field type `ptype`. 
2. Returned trajectory dict now uses `(self.ptype,'particle_index')` and `(self.ptype,'particle_time')` instead of just `'particle_index'` and `'particle_time'`

Change 2 should ensure the documentation for `trajectory_from_index` actually is correct (it plots using `("all","particle_time") which would not match what was actually returned).

Change 1 should fix Issue #4766.

Change 2 should ensure the documentation for `trajectory_from_index` actually is correct (it plots using `("all","particle_time") which would not match what was actually returned).

## PR Checklist

- [ ] New features are documented, with docstrings and narrative docs
- [x] Adds a test for any bugs fixed. Adds tests for new features.

### CODE REVIEW

**Comment by mtryan83** [comment]
I'm currently working on adding/updating the test(s). I'll try to update the test file to a pytest at the same time, but it involves converting two `yield` tests (see #3734), which I'm unfamiliar with.

**Comment by mtryan83** [comment]
Just comment: the `test_orbit_traj` and `test_etc_traj` don't seem to be running at all right now. They're excluded from the pytest framework and can't be found under the nose testing.
image
This seems to be true for a number of other tests too, like `test_pickling.test_particle_pickles`, so I don't know if it's intentional or not.

**Comment by neutrinoceros** [comment]
Hi @mtryan83, thanks for investigating this. To answer your question, I don't think that these tests are skipped intentionally, but as we haven't been running any kind of coverage analysis for a while, it's not surprising.
For the record I think your patch is sound as is but I'll wait until you're satisfied with testing to merge.

**Comment by mtryan83** [comment]
The tests probably aren't necessary, but I've already got them, so... 

I've included two tests, one to check that the default fields are set to tuples correctly and one to check that `particle_index` and `particle_time` are tuplized in the `trajectory_from_index` function. 

For each test I've included two versions, one designed for nose and one for pytest (currently commented out) that uses parameterize decorators and fixtures. Technically, the nose ones could also be used in pytest, they just don't use the fancy parts. I think it'd be better just to switch over when the changeover occurs.

I've kept test_particle_trajectories in the pytest ignore list for now, but it doesn't need to be. Since the two answer tests (`test_orbit_traj` and `test_etc_traj`) are still yield tests, pytest skips them. This seems ok to me, since they're currently broken anyway (per comment above and #4590).

**Comment by neutrinoceros** [comment]
Personally I would prefer that we only add the pytest version of these tests. You can do this by making them part of a new module (keep the name, just add a `_pytest` suffix), and adding this module to ignore lists for nose. One of these lists lives in `tests/tests.yaml` and the other one is `nose_ignores.txt`

**Comment by mtryan83** [comment]
I moved the `test_uniqueness` and `test_ptype` functions to `test_particle_trajectories_pytest.py` as well, since they work fine under pytest. This way the only nose tests are the two broken ones.

File: yt/data_objects/tests/test_particle_trajectories_pytest.py
Comment (neutrinoceros) [inline yt/data_objects/tests/test_particle_trajectories_pytest.py:123]: The use of parentheses seems to in indicate you meant to use a tuple here, I think ? The issue is that it's the not parentheses that make a tuple, it's really the comma ! in other words, `("io") == "io"`.
```suggestion
@pytest.mark.parametrize("ptype", [None, "io"])
```

File: yt/data_objects/tests/test_particle_trajectories_pytest.py
Comment (neutrinoceros) [inline yt/data_objects/tests/test_particle_trajectories_pytest.py:139]:
```suggestion
@pytest.mark.parametrize("ptype", [None, "io"])
```

File: yt/data_objects/tests/test_particle_trajectories_pytest.py
Comment (neutrinoceros) [inline yt/data_objects/tests/test_particle_trajectories_pytest.py:116]: Not new, but let's clean this up while we're at it
```suggestion
```

**Review by neutrinoceros — APPROVED** [review]
some very minor points, otherwise LGTM, thank you !

**Review by jzuhone — APPROVED** [review]

**Pull request merged**
